backend/Bindings/Edna.Bindings.LtiAdvantage/Services/NrpsClient.cs

```diff
@@ -111,5 +122,32 @@ public async Task<Member> GetById(string clientId, string tokenUrl, string audie
 
             return allMembers.FirstOrDefault(member => member.UserId.Equals(userId));
         }
+
+        private static string GetNextMembershipUrlFromHeaders(HttpResponseHeaders headers)
+        {
+            string linkUrls = headers.TryGetValues("Link", out var values) ? values.FirstOrDefault() : null;
+            if (string.IsNullOrEmpty(linkUrls))
+            {
+                return null;
+            }
+
+            string[] urls = linkUrls.Split(',');
+            string nextUrl = urls.FirstOrDefault(u => u.IndexOf("\"next\"", StringComparison.OrdinalIgnoreCase) >=0 );
+
+            if (string.IsNullOrEmpty(nextUrl))
+            {
+                return null;
+            }
+
+            // Regex to get the url from next url excluding < and >; characters.
+            Regex reg = new Regex("(?<=<)(.*?)(?=>;)");
+            Match matchResults = reg.Match(nextUrl);

```

Comment:
Looks like this method can throw. Should we guard it via a try catch? (from [Docs](https://docs.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.match?view=net-6.0#system-text-regularexpressions-regex-match(system-string)))